source/connection-monitoring-and-pooling/tests/README.rst

```diff
@@ -36,7 +36,22 @@ Unit Test Format:
 
 All Unit Tests have some of the following fields:
 
-- ``poolOptions``: if present, connection pool options to use when creating a pool
+- ``poolOptions``: If present, connection pool options to use when creating a pool;
+  both `standard ConnectionPoolOptions <https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#connection-pool-options-1>`__
+  and the following test-specific options are allowed:
+
+  - ``backgroundThreadIntervalMS``: A desired time interval between the end of a
+    `Background Thread Run <https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#background-thread>`__
+    and the beginning of the next Run. If a Connection Pool does not implement a Background Thread, the Test Runner MUST ignore the option.
+    If the option is not specified, an implementation is free to use any value it finds reasonable.
+    An implementation is also free to use an interval smaller than the desired one. Once the specified time interval since the end of a Run passes,
```

Comment:
> An implementation is also free to use an interval smaller than the desired one

Do we want to define Interval as the minimal interval between the runs? Or clarify why would implementation use smaller interval than specified.

Reply:
It is of the essence to ask and answer a question: how can such a restriction benefit the tests? As far as I know, so far such a benefit has not been expressed.

Here are the benefits of not specifying such a restriction:

1. The requirement _"Marking a pool as ready MUST start a new Run as soon as possible."_ requires violating the "minimal interval" restriction.
2. If a pool implementation wants to guarantee a Run while the pool is paused to clear resources, then having the "minimal interval" restriction complicates the implementation.
3. Without the "minimal interval" restriction, an implementation is free to implement the desired interval as desired period, a.k.a. frequency. Currently the Java driver implements it as desired frequency, and the corresponding Java API is intentionally public. Having the "minimal interval" restriction would require the Java driver to either change the specification of its public API, or introduce an additional internal API with a different implementation, which clogs up the implementation.

Reply:
Changed the wording.